### PR TITLE
Improve password reset flow

### PR DESCRIPTION
**Pull request by ranquild**
Epic: https://github.com/metabase/metabase/issues/19476

- Changes visuals
- Adds password validation
- Converts to typescript and styled components

Screenshot 2021-12-27 at 19 11 22
Screenshot 2021-12-27 at 19 12 01

How to test:
- Configure email in Metabase. You can use `maildev` for this https://github.com/metabase/metabase/blob/f94c3c96f69bcdd0b1f297553f94d02210b17218/frontend/test/__support__/e2e/helpers/e2e-email-helpers.js#L2
- Open `/auth/forgot_password` and submit the form with your email
- Click the link in the email and reset the password
- Make sure it was changed correctly

### CODE REVIEW

**Review by guilhermerenew-zz — APPROVED** [review]

File: frontend/src/metabase/components/form/CustomForm.jsx [hunk 1]
```diff
@@ -270,17 +270,17 @@ export const CustomFormFooter = (
     cancelTitle = t`Cancel`,
     onCancel,
     footerExtraButtons,
-    isReverse,
```
Comment (ranquild) [inline frontend/src/metabase/components/form/CustomForm.jsx:273]: Not used now (I added this prop a month ago)

**Review by guilhermerenew-zz — APPROVED** [review]

File: frontend/src/metabase/entities/users/forms.js [hunk 2]
```diff
@@ -127,6 +127,16 @@ export default {
       ...PASSWORD_FORM_FIELDS,
     ],
   },
+  password_forgot: {
+    fields: [
+      {
+        name: "email",
+        title: t`Email address`,
+        placeholder: "The email you use for your Metabase account",
```
Comment (kulyk) [inline frontend/src/metabase/entities/users/forms.js:135]: I believe we should translate the placeholder too

File: frontend/src/metabase/auth/containers/ResetPasswordApp/ResetPasswordApp.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { connect } from "react-redux";
+import ResetPassword from "../../components/ResetPassword";
+import {
+  resetPassword,
+  validatePassword,
+  validatePasswordToken,
+} from "metabase/auth/actions";
```
Comment (kulyk) [inline frontend/src/metabase/auth/containers/ResetPasswordApp/ResetPasswordApp.tsx:7]: Relative imports should go below the absolute ones by our style guide AFAIR

File: frontend/src/metabase/auth/components/AuthLayout/AuthLayout.styled.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  }
+
+  @media screen and (min-width: 1600px) {
+    left: -640px;
+  }
+
+  @media screen and (min-width: 1920px) {
+    left: 0;
+    width: 100%;
+  }
```
Comment (kulyk) [inline frontend/src/metabase/auth/components/AuthLayout/AuthLayout.styled.tsx:56]: I wonder if we can reuse shared media queries from [here](https://github.com/metabase/metabase/blob/master/frontend/src/metabase/styled-components/theme/media-queries.js)?

**Review by kulyk — APPROVED** [review]
Looks great! Have a few minor style-related comments and a translation thingie

File: frontend/src/metabase/entities/users/forms.js (hunk 2 above)
Comment (ranquild) [reply frontend/src/metabase/entities/users/forms.js]: Thanks, fixed!

File: frontend/src/metabase/auth/containers/ResetPasswordApp/ResetPasswordApp.tsx (hunk 3 above)
Comment (ranquild) [reply frontend/src/metabase/auth/containers/ResetPasswordApp/ResetPasswordApp.tsx]: Thanks, fixed!

File: frontend/src/metabase/auth/components/AuthLayout/AuthLayout.styled.tsx (hunk 4 above)
Comment (ranquild) [reply frontend/src/metabase/auth/components/AuthLayout/AuthLayout.styled.tsx:56]: It seems we cannot - these breakpoints are specific to the image :(

**Pull request merged**